cmd/panfs-multipart.go

```diff
@@ -354,7 +359,7 @@ func (fs *PANFSObjects) PutObjectPart(ctx context.Context, bucket, object, uploa
 	}
 
 	tmpPartPath := pathJoin(bucketPath, panfsS3TmpDir, fs.nodeDataSerial, uploadID+"."+mustGetUUID()+"."+strconv.Itoa(partID))
-	bytesWritten, err := fsCreateFile(ctx, tmpPartPath, data, data.Size())
+	bytesWritten, err := panfsCreateFile(ctx, tmpPartPath, data, data.Size(), fs.defaultObjMode, fs.defaultOwner, fs.defaultGroup)
```

Comment:
As we create upload part there and upload parts will always be written into target file and never renamed directly into the target location, I'd keep root permissions for them and avoid additional chwon syscalls.

Reply:
BTW, we create target file for multipart upload in ListObjectParts (see appendFilePath). And we need to make sure that this file has proper uid/gid setup. Maybe (this is just a proposal), we can rework only single function, which is our version or rename to:

- accept uid/gid/mode as input arguments [done]
- create destination hierarchy with provided mode/uid/gid [done]
- call chown on the dest file before rename
- perform rename (link + unlink) [done]

in that case, we may omit almost all changes in PanFS module (filesystem and multipart) except places where we rename something from temporary location to target location. And maybe it is just worth to call the function "publish()" instead of "rename()" to emphasize this specific, that we make minio temporary/internal file visible in the target location.

Reply:
I reworked this task, and also moved files as Dmitri asked, please have a look one more time